src/Templates/AbstractTemplateCompatibility.php

```diff
@@ -0,0 +1,157 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * BlockTemplatesCompatibility class.
+ *
+ * To bridge the gap on compatibility with PHP hooks and blockified templates.
+ *
+ * @internal
+ */
+abstract class AbstractTemplateCompatibility {
+	/**
+	 * The data of supported hooks, containing the hook name, the block name,
+	 * position, and the callbacks.
+	 *
+	 * @var array $hook_data The hook data.
+	 */
+	protected $hook_data;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->set_hook_data();
+		$this->init();
+	}
+
+	/**
+	 * Initialization method.
+	 */
+	protected function init() {
+		if ( ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		add_filter( 'render_block_data', array( $this, 'update_render_block_data' ), 10, 3 );
+		add_filter( 'render_block', array( $this, 'inject_hooks' ), 10, 2 );
+	}
```

Comment:
This is my mistake from the initial implementation. We shouldn't add hooks in the constructor of the class. Instead, we should fire the `init` method in the `Bootstrap` class instead.

```suggestion
	/**
	 * Initialization method.
	 */
	protected function init() {
		if ( ! wc_current_theme_is_fse_theme() ) {
			return;
		}

		$this->set_hook_data();

		add_filter( 'render_block_data', array( $this, 'update_render_block_data' ), 10, 3 );
		add_filter( 'render_block', array( $this, 'inject_hooks' ), 10, 2 );
	}
```